nifi-nar-bundles/nifi-iceberg-bundle/nifi-iceberg-processors/src/main/java/org/apache/nifi/processors/iceberg/PutIceberg.java

```diff
@@ -0,0 +1,252 @@
+            .allowableValues(
+                    new AllowableValue("AVRO"),
+                    new AllowableValue("PARQUET"),
+                    new AllowableValue("ORC"))
+            .required(false)
+            .build();
+
+    static final PropertyDescriptor TARGET_FILE_SIZE = new PropertyDescriptor.Builder()
+            .name("target-file-size")
+            .displayName("Target File Size")
+            .description("Controls the size of files generated to target about this many bytes. If not set the 'write.target-file-size-bytes' table property will be used, default value is 512 MB.")
+            .addValidator(StandardValidators.LONG_VALIDATOR)
+            .required(false)
+            .build();
+
+    static final Relationship REL_SUCCESS = new Relationship.Builder()
+            .name("success")
+            .description("A FlowFile is routed to this relationship after the data ingestion was successful.")
+            .build();
+
+    static final Relationship REL_FAILURE = new Relationship.Builder()
+            .name("failure")
+            .description("A FlowFile is routed to this relationship if the data ingestion failed and retrying the operation will also fail, such as an invalid data or schema.")
+            .build();
+
+    private static final List<PropertyDescriptor> PROPERTIES = Collections.unmodifiableList(Arrays.asList(
+            RECORD_READER,
+            CATALOG,
+            CATALOG_NAMESPACE,
+            TABLE_NAME,
+            FILE_FORMAT,
+            TARGET_FILE_SIZE,
+            KERBEROS_USER_SERVICE
+    ));
+
+    public static final Set<Relationship> RELATIONSHIPS = Collections.unmodifiableSet(new HashSet<>(Arrays.asList(
+            REL_SUCCESS,
+            REL_FAILURE
+    )));
+
+    @Override
+    protected List<PropertyDescriptor> getSupportedPropertyDescriptors() {
+        return PROPERTIES;
+    }
+
+    @Override
+    public Set<Relationship> getRelationships() {
+        return RELATIONSHIPS;
+    }
+
+    @Override
+    public void doOnTrigger(ProcessContext context, ProcessSession session) throws ProcessException {
+        FlowFile flowFile = session.get();
+        if (flowFile == null) {
+            return;
+        }
+
+        final RecordReaderFactory readerFactory = context.getProperty(RECORD_READER).asControllerService(RecordReaderFactory.class);
+        final String fileFormat = context.getProperty(FILE_FORMAT).evaluateAttributeExpressions().getValue();
+        final String targetFileSize = context.getProperty(TARGET_FILE_SIZE).evaluateAttributeExpressions().getValue();
+
+        Table table;
+
+        try {
+            table = loadTable(context);
+        } catch (Exception e) {
+            getLogger().error("Failed to load table from catalog", e);
+            session.transfer(flowFile, REL_FAILURE);
+            return;
+        }
+
+        final FileFormat format = getFileFormat(table.properties(), fileFormat);
```

Comment:
What happens when this method throws an exception? In a more severe scenario (failed to load table), the flow file will be routed to failure. In this situation, a process exception will be thrown when this method is run as a privileged action, and if not, the original exception will penalize the processor.